### DISCUSSION TITLE
i3-overview for workspace-heavy users

### DISCUSSION THREAD

**Discussion by milgra** [i3/i3#4700] [Show and tell]
Dear i3 folks,

I'm a long time user of i3 and I love it, but sometimes when I have more than 10 workspaces open I start to forget what is where and I start to switch workspaces blindly till I find what I want.

To solve this problem I created a simple app called i3-overview. It shows all workspaces at once, simple schematic drawings of windows in them with easily readable program names and titles, bound to the meta key press/release. It solved the problem for me, maybe someone else also finds it useful, feel free to try/use/improve it!

https://github.com/milgra/i3-overview

Regards,
Milan

**Comment by ghost** [comment]
Dang, that's a really cool idea!

**Comment by windowschips** [comment]
Hey, any updates on it? Seems the repository has been removed or privated. I would really like to try this.

**Reply by ghost** [reply]
I know this isn't the response you wanted, but here's my approach. It's straightforward and sort of typical for these window managers, but I basically just associate certain workspaces with certain activities. For me, ...

1. Terminals (mainly programming)
2. Miscellaneous Terminals
3. Video (MPV) / Audacity
4. Firefox
5. Unused
6. Unused
7. Thunderbird / VirtualBox
8. Unused
9. Alternative VirualBox Workspace
0. Music Player (MoC)

The unused ones are workspaces I rarely use, but are nonetheless available if needed.

This makes it *so* much easier to know where to look for what I want, because then I need only mentally juggle whats on 1 and 2 if I'm working on something. Switching between 1 and 2 is done in mere milliseconds, so even if I get lost in terminals when I'm working on something, I'm where I need to be *very* quickly. It only becomes a *minor* inconvenience if I start needing other workspace for terminals, but, at least these days, that's very rare.

The idea is that after a while of doing this, it becomes second nature as to where you look for things, because you already intuit that, for example, workspace 4 is for Firefox, rather than looking around for it. Think of it like a keyboard, in that you don't have to look around for H; you just know where it is, because your finger has gone to the same place for years upon years.

**Reply by okraits** [reply]
I use a different approach, but to each his own. I kind of use one workspace for each project or task I'm working on.

1. Firefox for communication: mail, company chat, matrix chat, etc.
2. Project A
3. Project B
4. Project C
...
10. Miscellaneous: IRC, terminals for other stuff

On each project/task workspace there's usually a terminal running neovim in tmux and a firefox window.
Like with your approach, I keep the projects always on the same workspaces so I can quickly switch to each using "muscle memory".
